Tests/MongoSwiftTests/DocumentTests.swift

```diff
@@ -330,10 +360,28 @@ final class DocumentTests: MongoSwiftTestCase {
 
                 // for cB input:
                 // native_to_bson( bson_to_native(cB) ) = cB
-                expect(Document(fromBSON: cBData).rawBSON).to(equal(cBData))
+                let docFromCB = Document(fromBSON: cBData)
+                expect(docFromCB.rawBSON).to(equal(cBData))
+
+                // test round tripping through documents
+                let testRoundTrip = {
+                    let nativeFromDoc = docFromCB.toArray()
```

Comment:
can you explain a little bit what's going on here? Why do we need to convert this to an array?

Reply:
This spec test is designed for drivers like Python that use a `dict` as a `Document` and only write actual BSON bytes when necessary. We actually store everything in a `bson_t`, so we don't have such an abstraction, but we do have a translation layer where bugs can occur.

When we read from the document to the array, we exercise our functions that retrieve the data from the `bson_t` and convert them to Swift types (i.e. `from(iterator:)`). Writing them back from the array to a `Document` exercises our `encode(to:DocumentStorage)` which converts Swift types to BSON data. If they both work properly, the original underlying `bson_t` should be identical to the one wrapped by the document created from the array, even though the types have been round tripped through our translation layer.

I added a comment explaining this more clearly.